pkgs/applications/virtualization/singularity/template.nix

```diff
@@ -0,0 +1,187 @@
+  ];
+
+  postPatch = ''
+    substituteInPlace internal/pkg/build/files/copy.go \
+      --replace /bin/cp ${coreutils}/bin/cp
+  '';
+
+  configureScript = "./mconfig";
+
+  configureFlags = [
+    "--localstatedir=/var"
+  ]
+  ++ lib.optional withoutSuid "--without-suid";
+
+  preConfigure = ''
+    if ! [ -e .git ] || ! [ -e VERSION ]; then
+      echo "${version}" > VERSION
+    fi
+  '';
+
+  postConfigure = ''
+    patchShebangs .
+
+    # Patching the hard-coded [Dd]efaultPath
+    for subPath in cmd/internal/cli/actions.go internal/pkg/util/env/env.go e2e/env/env.go; do
+      if [ -f "$subPath" ]; then
+        echo "Patching [Dd]efaultPath in \"$subPath\""
+        sed -i 's|^\([ \t]*[Dd]efaultPath[ \t]*[:]\?=[ \t]*\)"[^"]*"|\1"${lib.makeBinPath [ coreutils squashfsTools bash ]}"|' "$subPath"
+      fi
+    done
+
+    # Try to make use of the existing configureFlags and configureFlagsArray
+    # Code borrowed from pkgs/stdenv/generic/setup.sh configurePhase()
+
+    # set to empty if unset
+    : ''${configureScript=}
+    : ''${configureFlags=}
+
+    if [[ -z "''${dontAddPrefix:-}" && -n "$prefix" ]]; then
+        configureFlags="''${prefixKey:---prefix=}$prefix $configureFlags"
+    fi
+
+    # Old bash empty array hack
+    # shellcheck disable=SC2086
+    flagsArray=( $configureFlags "''${configureFlagsArray[@]}" )
+    echoCmd 'configure flags' "''${flagsArray[@]}"
+    # shellcheck disable=SC2086
+    $configureScript -V ${version} "''${flagsArray[@]}"
+    unset flagsArray
+
+    # End of the code from pkgs/stdenv/generic/setup.sh configurPhase
+
+    # Don't install SUID binaries
+    sed -i 's/-m 4755/-m 755/g' builddir/Makefile
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    make -C builddir
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    make -C builddir install LOCALSTATEDIR=$out/var
+
+    ${lib.optionalString (!withoutSuid) ''
+        chmod 755 $out/libexec/${projectName}/bin/starter-suid
+    ''}
+
+    # Explicitly configure paths in the config file
+    sed -i 's|^# mksquashfs path =.*$|mksquashfs path = ${lib.makeBinPath [squashfsTools]}/mksquashfs|' $out/etc/${projectName}/${projectName}.conf
+    sed -i 's|^# unsquashfs path =.*$|unsquashfs path = ${lib.makeBinPath [squashfsTools]}/unsquashfs|' $out/etc/${projectName}/${projectName}.conf
+    sed -i 's|^# cryptsetup path =.*$|cryptsetup path = ${lib.makeBinPath [cryptsetup]}/cryptsetup|' $out/etc/${projectName}/${projectName}.conf
+    sed -i 's|^# go path =.*$|go path = ${lib.makeBinPath [go]}/go|' $out/etc/${projectName}/${projectName}.conf
+    ${lib.optionalString enableNvidiaContainerCli ''
+      sed -i 's|^# nvidia-container-cli path =.*$|nvidia-container-cli path = ${lib.makeBinPath [nvidia-docker]}/nvidia-container-cli|' $out/etc/${projectName}/${projectName}.conf
+      sed -i 's|^# ldconfig path =.*$|ldconfig path = ${lib.makeBinPath [glibc.bin]}/ldconfig|' $out/etc/${projectName}/${projectName}.conf
```

Comment:
We should drop this (and the `glibc` argument): when Nix is used on top of a FHS distribution, it is explicitly expected that the underlying system's `ldconfig` would be used, cf. https://github.com/apptainer/singularity/pull/5671

Reply:
I'm also wondering if there could be an easier way to fix these paths than with `sed`.
I think `mconfig` automatically picks up things from `PATH` (and the current expression probably requires `build == host` anyway)

Reply:
I'll try that.

Reply:
> and the current expression probably requires `build == host` anyway

Sorry, but I'm not quite understand what `build == host` stands for. (I have seen similar conditional statement once, but I couldn't recall it.)

Reply:
I meant that the machine where we build the package and the machine where we use it probably (I'm really not familiar with the subject) should have the same platform because of how we use derivations (e.g. `nvidia-docker`) in string interpolation (e.g. `" ... ${... nvidia-docker ...} ..."`). If we just put these in `nativeBuildInputs`, we'd have the same limitation

Reply:
> We should drop this (and the `glibc` argument): when Nix is used on top of a FHS distribution, it is explicitly expected that the underlying system's `ldconfig` would be used, cf. [apptainer/singularity#5671](https://github.com/apptainer/singularity/pull/5671)

Done.

Reply:
> I think `mconfig` automatically picks up things from `PATH` (and the current expression probably requires `build == host` anyway)

Yes, but it somehow fails to pick up libraries we provide such as `nvidia-container-cli` and `cryptsetup`.